uvm/servlets/admin/config/network/Interface.js

```diff
@@ -85,7 +85,27 @@ Ext.define('Ung.config.network.Interface', {
                     },
                     minValue: 1,
                     maxValue: 4094,
-                    allowBlank: false
+                    allowBlank: false,
+                    validator: function(value) {
+                        var store = this.up('tabpanel').getViewModel().getStore('interfaces');
+                        var currentViewModel = this.up('window').getViewModel();
+
+                            // Convert value to integer
+                            var intValue = parseInt(value, 10);
+
+                            // Find records matching both vlanTag and vlanParent
+                            var matchedRecords = store.queryBy(function(record) {
+                                return record.get('vlanTag') === intValue && record.get('vlanParent') === currentViewModel.get('intf.vlanParent');
+                            });
+
+                            if (matchedRecords.getCount() > 0) {
+                                var matchedRecord = matchedRecords.first(); 
+                                if (matchedRecord.get('name') !== currentViewModel.get('intf.name')) {
+                                    return 'VLAN Tag ' + value + ' is already matched with interface ' + matchedRecord.get('name');
```

Comment:
Look into ExtJs's Ext.String.format function for formatting strings without breaking them up/concatenating.